ACE/tests/Message_Queue_Test_Ex.cpp

```diff
@@ -709,6 +709,98 @@ int queue_priority_test (ACE_Message_Queue_Ex<User_Class, ACE_SYNCH>& q)
   return status;
 }
 
+class Queue_Ex_Iterator_No_Lock
+ : public ACE_Message_Queue_Iterator<ACE_SYNCH, ACE_System_Time_Policy>
+{
+  typedef ACE_Message_Queue_Iterator<ACE_SYNCH, ACE_System_Time_Policy> inherited;
+
+ public:
+  typedef ACE_Message_Queue_Ex<User_Class, ACE_SYNCH, ACE_System_Time_Policy> MESSAGE_QUEUE_EX_T;
+
+  explicit Queue_Ex_Iterator_No_Lock (MESSAGE_QUEUE_EX_T& queue_in)
+   : inherited (queue_in.queue ())
+  {}
+  virtual ~Queue_Ex_Iterator_No_Lock () {}
```

Comment:
Should be `default`, use `override` instead of virtual

Reply:
ok. I don't know how to use 'override'; does not seem to work. Can you give an example ?

Reply:
https://en.cppreference.com/w/cpp/language/override

Reply:
if I do `~Queue_Ex_Iterator_No_Lock () override = default;` I get: 

> method with override specifier 'override' did not override any base class methods

Reply:
Ok, than only use default, remove virtual, the base class has a non-virtual destructor. Maybe you can add the default destructor also to the base?

Reply:
It looks it is not really intended to inherit, why not remove the inheritance, you already implement most methods in this class already

Reply:
You are right, but I cannot do this because I need to reference its protected member `curr_`, and the base class needs to be a `friend` of the queue so it can set `curr_`. Otherwise my class would also need to be a `friend` of the queue, which would also be very ugly design